interfaces/console/src/app/console/sites/page.tsx

```diff
@@ -92,345 +91,134 @@ export default function Page() {
     },
   });
 
-  const [getSiteMetrics, { variables: getSiteMetricsVar }] =
-    useGetSiteStatLazyQuery({
+  const [getSiteMetrics, { loading: metricsLoading }] = useGetSiteStatLazyQuery(
+    {
       client: subscriptionClient,
       fetchPolicy: 'network-only',
       onCompleted: (data) => {
+        console.log('Initial site metrics received:', data);
         if (data.getSiteStat.metrics.length > 0) {
-          const siteId = getSiteMetricsVar?.data.siteId || '';
+          const siteId = data.getSiteStat.metrics[0].siteId;
           const metrics = data.getSiteStat.metrics;
 
+          let updatedSiteMetrics: Partial<SiteMetrics> = {};
+
           metrics.forEach((metric) => {
-            if (metric.type === 'site_uptime_seconds') {
-              setSitesStatus((prev) => {
-                const currentStatus = prev[siteId] || {
-                  status: SITE_STATUS.ONLINE,
-                  batteryStatus: 'Charged',
-                  signalStrength: 'Strong',
-                };
-
-                return {
-                  ...prev,
-                  [siteId]: {
-                    ...currentStatus,
-                    status:
-                      metric.value <= 0
-                        ? SITE_STATUS.OFFLINE
-                        : currentStatus.status,
-                  },
-                };
-              });
+            switch (metric.type) {
+              case 'site_uptime_seconds':
+                updatedSiteMetrics.siteUptimeSeconds = metric.value;
+                break;
+              case 'battery_charge_percentage':
+                updatedSiteMetrics.batteryPercentage = metric.value;
+                break;
+              case 'backhaul_speed':
+                updatedSiteMetrics.backhaulSpeed = metric.value;
+                break;
             }
           });
 
-          // Set up subscription for real-time updates
-          const sKey = `stat-${user.orgName}-${user.id}-${Stats_Type.Site}-${siteId}`;
-          MetricStatSubscription({
-            key: sKey,
-            siteId: siteId,
-            userId: user.id,
-            url: env.METRIC_URL,
-            orgName: user.orgName,
-            type: Stats_Type.Site,
-            from: getUnixTime() - 40, // 24 hours
+          setSiteMetrics((prev) => {
+            const updated = {
+              ...prev,
+              [siteId]: {
+                ...prev[siteId],
+                ...updatedSiteMetrics,
+              },
+            };
+            persistedSiteMetrics = updated;
+            return updated;
           });
```

Comment:
managing state for metrics looks an overhead, we already have PubSub behaviour on console for metrics. Rather doing props drilling we can publish the metrics value against unique key and subscribe to same key on a component.